src/main/java/de/tum/in/www1/artemis/config/migration/setups/localvc/gitlab/MigrationEntryGitLabToLocalVC.java

```diff
@@ -0,0 +1,321 @@
+     * @param programmingExercise   the programming exercise to migrate the auxiliary repositories for
+     * @param oldBranch             the old branch of the programming exercise
+     */
+    private void migrateAuxiliaryRepositories(SolutionProgrammingExerciseParticipation solutionParticipation, ProgrammingExercise programmingExercise, String oldBranch) {
+        for (var repo : getAuxiliaryRepositories(programmingExercise.getId())) {
+            try {
+                if (repo.getRepositoryUri() == null) {
+                    log.error("Repository URI is null for auxiliary repository with id {}, cant migrate", repo.getId());
+                    continue;
+                }
+                var url = cloneRepositoryFromSourceVCSAndMoveToLocalVC(programmingExercise, repo.getRepositoryUri(), oldBranch);
+                if (url == null) {
+                    errorList.add(solutionParticipation);
+                    log.error("Failed to migrate auxiliary repository for programming exercise with id {}, keeping the url in the database", programmingExercise.getId());
+                }
+                else {
+                    log.debug("Migrated auxiliary repository with id {} to {}", repo.getId(), url);
+                    repo.setRepositoryUri(url);
+                    auxiliaryRepositoryRepository.save(repo);
+                }
+            }
+            catch (Exception e) {
+                log.error("Failed to migrate auxiliary repository with id {}", repo.getId(), e);
+            }
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void migrateSolutions(List<SolutionProgrammingExerciseParticipation> solutionParticipations) {
+        for (var solutionParticipation : solutionParticipations) {
+            try {
+                if (isRepositoryUriNotNull(solutionParticipation, "Repository URI is null for solution participation with id {}, cant migrate")) {
+                    var programmingExercise = solutionParticipation.getProgrammingExercise();
+                    var url = cloneRepositoryFromSourceVCSAndMoveToLocalVC(solutionParticipation.getProgrammingExercise(), solutionParticipation.getRepositoryUri(),
+                            programmingExercise.getBranch());
+                    if (url == null) {
+                        log.error("Failed to migrate solution repository for solution participation with id {}, keeping the url in the database", solutionParticipation.getId());
+                        errorList.add(solutionParticipation);
+                    }
+                    else {
+                        log.debug("Migrated solution participation with id {} to {}", solutionParticipation.getId(), url);
+                        solutionParticipation.setRepositoryUri(url);
+                        solutionProgrammingExerciseParticipationRepository.save(solutionParticipation);
+                    }
```

Comment:
```suggestion
                    }

                    url = migrateTestRepo(solutionParticipation.getProgrammingExercise());
```
nitpick: giving the code a bit of breathing space to mark that solution migration is done, now test migrations will happen makes it easier to read.